crates/rbuilder/src/utils/provider_factory_reopen.rs

```diff
@@ -61,13 +71,23 @@ impl<DB: Database + Clone> ProviderFactoryReopener<DB> {
     /// This will check if historical block hashes for the given block is correct and if not it will reopen
     /// provider fatory.
     /// This should be used when consistency is required: e.g. building blocks.
-    pub fn check_consistency_and_reopen_if_needed(
-        &self,
-        current_block_number: u64,
-    ) -> eyre::Result<ProviderFactory<DB>> {
+    ///
+    /// If the current block number is already known at the time of calling this method, you may pass it to
+    /// avoid an additional DB lookup for the latest block number.
+    pub fn check_consistency_and_reopen_if_needed(&self) -> eyre::Result<ProviderFactory<DB>> {
+        let best_block_number = self
+            .provider_factory_unchecked()
+            .best_block_number()
+            .map_err(|err| eyre::eyre!("Error getting best block number: {:?}", err))?;
         let mut provider_factory = self.provider_factory.lock().unwrap();
-        if !self.testing_mode {
-            match check_provider_factory_health(current_block_number, &provider_factory) {
+
+        // Don't need to check consistency for the block that was just checked.
```

Comment:
I am pretty sure that its true but it can be possible that its not.

The only place that I see that would suffer from too much checking here is nonce cache in sim tree. But even then this check can have a small impact.

I don't think that we should change it now but I marked it to remember to revisit if we see some problems with reopening.